service/controller/resource/tccpn/template/template_main_auto_scaling_group.go

```diff
@@ -25,18 +26,21 @@ const TemplateMainAutoScalingGroup = `
       - {{ .LoadBalancers.ApiInternalName }}
       - {{ .LoadBalancers.ApiName }}
       - {{ .LoadBalancers.EtcdName }}
-
+     {{ if $HAMasters }}
+      # We define lifecycle hook only in case of HA masters.
+      # in case of 1 masters the draining would not work as the API is down when we try to roll the instance.
+      #
       # We define a lifecycle hook as part of the ASG in order to drain nodes
       # properly on Node Pool deletion. Earlier we defined a separate lifecycle
       # hook referencing the ASG name. In this setting when deleting a Node Pool
       # the lifecycle hook was never executed. We always want node draining for
       # reliably managing customer workloads.
       LifecycleHookSpecificationList:
         - DefaultResult: CONTINUE
-          HeartbeatTimeout: 3600
+          HeartbeatTimeout: 600
```

Comment:
This sets the timeout to 10 minutes if I am not mistaken. We just talked about the problems of not having enough time during incidents and failover situations. I think we should rather increase the timeout instead of lowering it. What was your idea with this change?

Reply:
Yah this was early change to quicken the deletion process, this will go back to 3600